### PR TITLE
Feat: Implement OBProxy Management module

### PR DESCRIPTION
**Pull request by powerfooI**
## Summary

1. Adjusted `distribution/obproxy/start.sh`. It now read env vars which have prefix `ODP_` and set them as startup options.
2. Implemented APIs for OBProxy and tested them.
3. Implemented OBProxy monitoring.

### CODE REVIEW

File: internal/dashboard/business/obproxy/configmap.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			Namespace: ns,
+			Labels: map[string]string{
+				LabelOBProxy:            name,
+				constant.LabelManagedBy: constant.DASHBOARD_APP_NAME,
+			},
+		},
+		Data: map[string]string{},
+	}
+	for _, kv := range param.Parameters {
+		cmParam.Data[strings.ToUpper(envPrefix+kv.Key)] = kv.Value
```
Comment (chris-sun-star) [inline internal/dashboard/business/obproxy/configmap.go:54]: why not use the original key, convert key to upper and lower may cause inconsistency

File: internal/dashboard/business/obproxy/const.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+const (
+	envPrefix = "ODP_"
+)
+
+const (
+	cmSuffix             = "-odp-config"
+	svcSuffix            = "-odp-svc"
+	proxyRoSecretSuffix  = "-proxyro-secret"
+	proxySysSecretSuffix = "-proxysys-secret"
```
Comment (chris-sun-star) [inline internal/dashboard/business/obproxy/const.go:23]: the suffix is too long, will limit the length of obproxy name

File: internal/dashboard/model/obproxy/param.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+package obproxy
+
+import "github.com/oceanbase/ob-operator/internal/dashboard/model/common"
+
+type CreateOBProxyParam struct {
+	Name             string `json:"name" binding:"required"`
+	Namespace        string `json:"namespace" binding:"required"`
+	ProxyClusterName string `json:"proxyClusterName" binding:"required"`
+	// OBCluster format: {namespace}/{name}
+	OBCluster string `json:"obCluster" binding:"required"`
```
Comment (chris-sun-star) [inline internal/dashboard/model/obproxy/param.go:22]: Pass obcluster in a structured item

File: internal/dashboard/model/obproxy/response.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	Resource       common.ResourceSpec   `json:"resource" binding:"required"`
+	Parameters     []common.KVPair       `json:"parameters" binding:"required"`
+	Pods           []response.K8sPodInfo `json:"pods" binding:"required"`
+}
+
+type OBProxyOverview struct {
+	Name      string `json:"name" binding:"required"`
+	Namespace string `json:"namespace" binding:"required"`
+	// OBCluster format: {namespace}/{name}
+	OBCluster        string `json:"obCluster" binding:"required"`
```
Comment (chris-sun-star) [inline internal/dashboard/model/obproxy/response.go:34]: define obcluster in a structured item

File: internal/dashboard/business/obproxy/obproxy.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		}
+		return nil, httpErr.NewInternal("Failed to get obcluster, err msg: " + err.Error())
+	}
+	// TODO: Decrypted the password
+	proxySysSecret, err := createPasswordSecret(ctx, param.Namespace, param.Name+proxySysSecretSuffix, param.ProxySysPassword)
+	if err != nil {
+		return nil, err
+	}
+	var proxyRoSecret *corev1.Secret
+	if param.Namespace != clusterNs {
```
Comment (chris-sun-star) [inline internal/dashboard/business/obproxy/obproxy.go:77]: copy the secret no matter if the namespaces are the same

File: distribution/obproxy/start.sh [hunk 6]
```diff
@@ -1,21 +1,57 @@
...
     echo "env variable APP_NAME is required"
     exit 1
 fi
 
 if [ -z $PROXYRO_PASSWORD_HASH ]; then
-    PROXYRO_PASSWORD_HASH=`echo -n "$PROXYRO_PASSWORD" | sha1sum | awk '{print $1}'`
+    PROXYRO_PASSWORD_HASH=$(echo -n "$PROXYRO_PASSWORD" | sha1sum | awk '{print $1}')
 fi
 
+if [ -z $PROXYSYS_PASSWORD ]; then
```
Comment (chris-sun-star) [inline distribution/obproxy/start.sh:12]: PROXYSYS_PASSWORD can be empty

File: distribution/obproxy/start.sh [hunk 7]
```diff
@@ -1,21 +1,57 @@
...
+    echo "env variable PROXYSYS_PASSWORD is required"
+    exit 1
+fi
+
+PROXYSYS_PASSWORD_HASH=$(echo -n "$PROXYSYS_PASSWORD" | sha1sum | awk '{print $1}')
```
Comment (chris-sun-star) [inline distribution/obproxy/start.sh:17]: calculate PROXYSYS_PASSWORD_HASH only when it's empty

File: distribution/obproxy/start.sh [hunk 8]
```diff
@@ -1,21 +1,57 @@
...
+[ -z "$ODP_LOG_DIR_SIZE_THRESHOLD" ] && opts=$(concat_opts $opts "log_dir_size_threshold=10G")
+[ -z "$ODP_ENABLE_PROXY_SCRAMBLE" ] && opts=$(concat_opts $opts "enable_proxy_scramble=true")
+[ -z "$ODP_ENABLE_STRICT_KERNEL_RELEASE" ] && opts=$(concat_opts $opts "enable_strict_kernel_release=false")
+
+while IFS='=' read -r key value; do
+    # If the key has prefix "ODP_" then add it to the opts
+    if [[ $key == ODP_* ]]; then
+        # Remove the prefix "ODP_" from the key and transform to lower case
+        key=$(echo $key | sed 's/^ODP_//g' | tr '[:upper:]' '[:lower:]')
+        opts=$(concat_opts $opts "$(printf "%s=%s" "$key" "$value")")
```
Comment (chris-sun-star) [inline distribution/obproxy/start.sh:43]: it's better to add a quote for value, some config values may contain special characters such as '&' which will cause the command behave unexpectedly

Comment (powerfooI) [reply distribution/obproxy/start.sh:43]: You mean `opts=$(concat_opts $opts "$(printf "%s=\"%s\"" "$key" "$value")")`?

Comment (chris-sun-star) [reply distribution/obproxy/start.sh]: yes

**Review by chris-sun-star — APPROVED** [review]

**Pull request merged**